File: jenjin-core/src/main/java/com/binarymonks/jj/pools/managers/Vector3PoolManager.java
package com.binarymonks.jj.pools.managers;

import com.badlogic.gdx.math.Vector3;
import com.binarymonks.jj.pools.PoolManager;

public class Vector3PoolManager implements PoolManager<Vector3> {
    @Override
    public void reset(Vector3 vector3) {
        vector3.set(0, 0, 0);
    }

    @Override
    public Vector3 create_new() {
        return new Vector3();
    }

    @Override
    public void dispose(Vector3 vector3) {

    }
}

File: jenjin-core/src/main/kotlin/com/binarymonks/jj/core/pools/managers/Vector3PoolManager.kt
package com.binarymonks.jj.core.pools.managers

import com.badlogic.gdx.math.Vector3
import com.binarymonks.jj.core.pools.PoolManager

class Vector3PoolManager : PoolManager<Vector3> {
    override fun reset(vector3: Vector3) {
        vector3.set(0f, 0f, 0f)
    }

    override fun create_new(): Vector3 {
        return Vector3()
    }

    override fun dispose(vector3: Vector3) {

    }
}
